client/src/main/java/com/paypal/selion/platform/grid/Grid.java

```diff
@@ -81,10 +81,15 @@ public static long getExecutionTimeoutValue() {
      * @return A {@link RemoteWebDriver} object which can be used in the case of {@link MobileTest} and {@link WebTest}s
      */
     public static RemoteWebDriver driver() {
-        if (! getTestSession().isStarted()) {
-            getTestSession().startSesion();
+        AbstractTestSession testSession = getTestSession();
+        if (! testSession.isStarted()) {
+            testSession.startSesion();
+        }
+        RemoteWebDriver rwd = threadLocalWebDriver.get();
+        if (rwd == null) {
+            throw new IllegalStateException("Driver not initialized. Is @WebTest annotation missing on method?");
```

Comment:
or `@MobileTest`.. If the user is running session sharing, the annotation will be on the class vs. the method.